ci/templates/ci/reports.html

```diff
@@ -100,42 +120,64 @@ <h3 class="reports__heading">Report Tasks</h3>
-        $.each(emails,function(i){
-          $("#id_email_to").after($(".email-to .email-to-group").clone().addClass("field-clone"));
-          $(".field-clone .email-fields").each(function(i){
-            $(this).attr('value', emails[i]);
+      // Replace encoded quotes with double quotes to make valid json
+      var forms = JSON.parse("{{ forms.keys }}".replace(/&#39;/g, '\"'))
+
+      $.each(forms, function(k){
+        // Get form and email_to field ids
+        var form_id = "#".concat(forms[k]);
+        var prefix = forms[k].replace("_form", "");
+        var email_to_id = "#".concat(prefix, "_email_to");
+
+        if ($(email_to_id).val() != '') {
+          var emails = $(email_to_id).val().split(",");
+          $(email_to_id).val('');
+          $.each(emails,function(i){
+            $(email_to_id).after($(".email-to .email-to-group").clone().addClass("field-clone"));
+            $(form_id).find(".field-clone .email-fields").each(function(i){
+              $(this).attr('value', emails[i]);
+            });
           });
-        });
         }
-      else {
-        $("#id_email_to").after($(".email-to .email-to-group").clone().addClass("field-clone"));
-      };
+        else {
+          $(email_to_id).after($(".email-to .email-to-group").clone().addClass("field-clone"));
+        };
 
-      $("#id_email_to").addClass("hide")
-      $(".field-clone:first .remove").replaceWith($(".email-to .add-more").show());
+        $(email_to_id).addClass("hide")
+        $(form_id).find(".field-clone:first .remove")
+            .replaceWith($(".email-to .add-more").clone().show());
+      });
 
       $(".add-more").click(function(){
-          $(".field-clone:last").after($(".email-to .email-to-group").clone());
+          $(this).parents(".control-group").after($(".email-to .email-to-group").clone());
       });
 
       $("body").on("click",".remove",function(){
           $(this).parents(".control-group").remove();
       });
 
       $("body").on("click",".generate",function(){
-        $(".form-horizontal .email-fields").each(function(){
+        var id_prefix = $(this).parents(".form-horizontal").attr("id").replace("_form", "");
+        $(this).parents(".form-horizontal").find(".email-fields").each(function(){
           if ($(this).val() == '')
             $(this).parents(".form-horizontal .control-group").remove();
-            $("#id_email_to").val($(".form-horizontal .email-fields").map( function() {
+            $("#".concat(id_prefix, "_email_to")).val($(".form-horizontal .email-fields").map( function() {
                 return this.value;
               }).get().join(","));
         });
       });
 
+      $("#report_type").on("change", function(){
+        var dropdown = $("#report_type");
+        var form = $("#"+dropdown.val());
+        $(".report-form:visible").hide();
+        form.show();
+        dropdown.attr("form", dropdown.val());
+      });
+
+      $("#report_type").val("{{ report_type }}");
```

Comment:
Can reuse local form var  already defined above

Reply:
So the form var defined higher up holds all the forms that should be available on the page.
`{{ report_type }}` on the other hand will only be populated if the user tried to submit a form and it will hold the name of the form that was submitted
My goal behind this line and the one below it is to ensure that if a user submits a form that form will display on the page after the reload. Otherwise it gets hidden again and the user has to reselect it from the dropdown to see any errors. Is there a better way to do this?

Reply:
👍